## 主入口 「 main.js 」

```js
import Koa from 'koa'
import { errorHandler } from './errorHandler.js'

// Koa 继承自 EventEmitter，可以直接触发和监听自定义事件
const app = new Koa()

errorHandler(app)

app.use(async (ctx, next) => {
  app.emit('error', -1001, ctx)
})

app.listen(3000)
```



## 错误处理 「 errorHandler.js 」

```js
export function errorHandler(app) {
  app.on('error', (errCode, ctx) => {
    let message = ''

    switch (errCode) {
      case -1001:
        message = '用户未登录'
        break
      case -1002:
        message = '用户已存在'
        break
      default:
        message = '未知错误'
    }

    ctx.body = {
      code: errCode,
      message
    }
  })
}
```

